core-bundle/contao/classes/BackendTemplate.php

```diff
@@ -88,7 +88,7 @@ private function compile()
 			foreach (array_unique($GLOBALS['TL_JAVASCRIPT']) as $javascript)
 			{
 				$options = StringUtil::resolveFlaggedUrl($javascript);
-				$strJavaScripts .= Template::generateScriptTag($this->addStaticUrlTo($javascript), $options->async, $options->mtime);
+				$strJavaScripts .= preg_replace('/>$/', ' data-turbo-track="dynamic">', Template::generateScriptTag($this->addStaticUrlTo($javascript), $options->async, $options->mtime));
```

Comment:
This cannot work I think as removing scripts won't unload them/remove their listeners/free their memory etc.

Reply:
```suggestion
				$strJavaScripts .= Template::generateScriptTag($this->addStaticUrlTo($javascript), $options->async, $options->mtime);
```

Imo that change is not needed as we are only talking about Styles in the Issue. We would not need to add the attribute to any JS.

Reply:
> as we are only talking about Styles in the Issue

However, the problem is triggered by dynamic scripts, too.

Reply:
> This cannot work I think as removing scripts won't unload them/remove their listeners/free their memory etc.

But that does not really matter, does it? In Contao 5.4+ your JavaScript needs to be compatible with this fact in any case. It does not matter whether you add a JavaScript [globally](https://docs.contao.org/dev/guides/adding-back-end-assets/#adding-assets-globally) or just in a [specific DCA](https://docs.contao.org/dev/guides/adding-back-end-assets/#adding-assets-for-a-dca) - in both cases your JavaScript needs to be compatible with the fact that Turbo will only render the `body` for example. You cannot rely on our `data-turbo-track` hash. The JavaScript cannot rely that listeners will automatically be removed when switching to another page, it cannot rely on `DOMContentLoaded` etc.

`data-turbo-track="dynamic"` on `<script>` makes sure that, if you have 100 extensions who add 100 different JavaScripts under 100 different URLs and you visit those 100 different URLs, you won't end up with the `<head>` being populated constantly with 100 different `<script>` tags in your DOM.

Reply:
> To quote the documentation: `Typically used to [remove style and link elements](https://turbo.hotwired.dev/handbook/drive#removing-assets-when-they-change) during navigation.`
> 
> https://turbo.hotwired.dev/handbook/drive#removing-assets-when-they-change
> 
> I do think it's the right fix but we should not add `dynamic` to JS.

@zoglo it says "typically used". Turbo just makes you aware how JavaScript needs to work with Turbo and wants to make sure that you don't get a misconception about this.